lib/charms/layer/jenkins/configuration.py

```diff
@@ -17,7 +17,19 @@ def bootstrap(self):
         hookenv.log("Bootstrapping initial Jenkins configuration")
 
         config = hookenv.config()
-        context = {"master_executors": config["master-executors"]}
+
+        # check for sane port values, fall back to the default
+        if not -1 <= config["jnlp-port"] <= 65535: # pragma: no cover
```

Comment:
I'd rather want to see a unit test for this situation, rather than a ``` # pragma: no cover```. Please have a look at unit_tests/test_configuration.py, it should be easy to a variation of test_bootstrap that exercises this code path. Also you should add a new assertion to test_bootstrap checking that the rendered file actually contains the expected XML node.